doi-transformer/src/test/java/no/unit/nva/doi/transformer/CrossRefConverterTest.java

```diff
@@ -118,46 +126,99 @@
     private static final String SAMPLE_ORCID = "http://orcid.org/0000-1111-2222-3333";
     private static final Instant SAMPLE_CROSSREF_DATE_AS_INSTANT = sampleCrossrefDateAsInstant();
     private static final ObjectMapper objectMapper = JsonUtils.objectMapper;
-    public static final int EXPECTED_NUMBER_OF_CONTRIBUTORS = 4;
+    private static final String ALTERNATIVE_TITLE = "Some alternative title";
+    private static final int SECOND_LIST_ELEMENT = 1;
     private final CrossRefConverter converter = new CrossRefConverter();
 
+    @Test
+    public void toPublicationReturnsObjectWithSourceLinkBeingTheFirstAvailableLinkToTheActualPublication() {
+        CrossRefDocument crossRefDocument = sampleBook();
+        assertThat(crossRefDocument.getLink(), is(not(empty())));
+        Publication publication = toPublication(crossRefDocument);
+        String actualSourceUriString = publication.getLink().toString();
+        String expectedSourceUriString = crossRefDocument.getLink().get(FIRST_LIST_ELEMENT).getUrl();
+        assertThat(actualSourceUriString, is(equalTo(expectedSourceUriString)));
+    }
+
+    @Test
+    public void toPublicationLogsWarningWhenCrossrefDocumentLinkContainsUriThatIsNotUrl() {
+        TestAppender logAppender = LogUtils.getTestingAppenderForRootLogger();
+        String invalidUrl = "uri:someUriThatIsNotUrl";
+        CrossRefDocument crossRefDocument = crossRefDocumentWithInvalidUrl(invalidUrl);
+        toPublication(crossRefDocument);
+        assertThat(logAppender.getMessages(), containsString(MALFORMED_URL_MESSAGE));
+        assertThat(logAppender.getMessages(), containsString(invalidUrl));
+    }
+
+    @Test
+    public void toPublicationLogsWarningWhenCrossrefDocumentLinkContainsInvalidUri() {
+        TestAppender logAppender = LogUtils.getTestingAppenderForRootLogger();
+        String invalidUri = "not a uri";
+        CrossRefDocument crossRefDocument = crossRefDocumentWithInvalidUrl(invalidUri);
+        toPublication(crossRefDocument);
+        assertThat(logAppender.getMessages(), containsString(invalidUri));
```

Comment:
Not an error message, just the invalid url value?

Reply:
The exception comes from the method URI.create, I am do not create a custom message.